### ISSUE TITLE
Failed to detect feed type on https://storyfm.cn/feed/episodes

### ISSUE THREAD

**Issue by gozelus** [mmcdole/gofeed#170]
```go
resp, _ := http.Get("https://storyfm.cn/feed/episodes")
t := gofeed.DetectFeedType(resp.Body)
fmt.Println(t)
```
### Expected behavior
it prints `0`
### Actual behavior
should print `FeedTypeRSS`

--------------
```go
gofeed.NewParser().ParseURL("https://storyfm.cn/feed/episodes")
```
 return error, but 
```go
resp, _ := http.Get("https://storyfm.cn/feed/episodes")
f, err := rss.Parser{}.Parse(resp.Body)
```
works fine.

**Comment by gozelus** [comment]
I have found the reason, the HTML file contains the bom lead to parse failure.
Is it possible to solve this problem?

**Comment by makeworld-the-better-one** [comment]
I believe the master branch has a fix for this, see #158 and https://github.com/mmcdole/gofeed/commit/d5ae83954e0552f161c04d5d1835d20f82950916.

**Comment by gozelus** [comment]
> I have found the reason, the HTML file contains the bom lead to parse failure.
> Is it possible to solve this problem?

It fixed, thanks so much !
